### PR TITLE
Add support for `applicationCategory` and `bundleDisplayName` helpers on `SettingsDictionary` to make it easier to fill up the identity section in Xcode

### PR DESCRIPTION
**Pull request by TheInkedEngineer**
### Short description 📝

Xcode recently introduced an identity section under General when the target is an Application. It includes fields that Tuist already cover like `BundleIdentifier`, `Version` and `Build`. Tuist however does not cover the `App Category` and `Display Name` which is the reason for this Pull Request.

### How to test the changes locally 🧐

Create a tuist project that include the two introduced settings, the `App Category` and `Display Name` under `Identity` in `General` should be populated.

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by TheInkedEngineer** [comment]
Currently in Draft until #5095 is merged.

File: Sources/ProjectDescription/SettingsTransformers.swift [hunk 1]
```diff
@@ -169,6 +212,18 @@ extension SettingsDictionary {
     public func debugInformationFormat(_ format: DebugInformationFormat) -> SettingsDictionary {
         merging(["DEBUG_INFORMATION_FORMAT": SettingValue(format)])
     }
+
+    // MARK: - Info.plist Values
+
+    /// Sets `INFOPLIST_KEY_LSApplicationCategoryType` to `category`.
```
Comment (danyf90) [inline Sources/ProjectDescription/SettingsTransformers.swift:218]:
```suggestion
    /// Sets `INFOPLIST_KEY_LSApplicationCategoryType` to the given category.
```

File: Sources/ProjectDescription/SettingsTransformers.swift [hunk 2]
```diff
@@ -169,6 +212,18 @@ extension SettingsDictionary {
...
+    public func applicationCategory(_ category: ApplicationCategory) -> SettingsDictionary {
+        merging(["INFOPLIST_KEY_LSApplicationCategoryType": SettingValue(category.rawValue)])
+    }
+
+    /// Sets `INFOPLIST_KEY_LSApplicationCategoryType` to `name`.
```
Comment (danyf90) [inline Sources/ProjectDescription/SettingsTransformers.swift:223]:
```suggestion
    /// Sets `INFOPLIST_KEY_CFBundleDisplayName` to the given name.
```

File: Tests/ProjectDescriptionTests/SettingsTests.swift [hunk 3]
```diff
@@ -281,4 +281,377 @@ final class SettingsTests: XCTestCase {
...
+        ])
+
+        /// Given/When
+        let settings40 = SettingsDictionary()
+            .applicationCategory(.weather)
+
+        /// Then
+        XCTAssertEqual(settings40, [
+            "INFOPLIST_KEY_LSApplicationCategoryType": "public.app-category.weather",
+        ])
```
Comment (danyf90) [inline Tests/ProjectDescriptionTests/SettingsTests.swift:655]: I wouldn't test every single value
```suggestion
```

**Review by danyf90 — APPROVED** [review]

File: Sources/ProjectDescription/SettingsTransformers.swift [hunk 4]
```diff
@@ -169,6 +212,18 @@ extension SettingsDictionary {
     public func debugInformationFormat(_ format: DebugInformationFormat) -> SettingsDictionary {
         merging(["DEBUG_INFORMATION_FORMAT": SettingValue(format)])
     }
+
+    // MARK: - Info.plist Values
+
+    /// Sets `INFOPLIST_KEY_LSApplicationCategoryType` to the given category.
+    public func applicationCategory(_ category: ApplicationCategory) -> SettingsDictionary {
+        merging(["INFOPLIST_KEY_LSApplicationCategoryType": SettingValue(category.rawValue)])
```
Comment (danyf90) [inline Sources/ProjectDescription/SettingsTransformers.swift:220]: Shouldn't this just be 
```suggestion
        merging(["LSApplicationCategoryType": SettingValue(category.rawValue)])
```
?
If we want it in the plist we should change the PLIST helper instead

Comment (kwridan) [reply Sources/ProjectDescription/SettingsTransformers.swift:220]: 👍 Tuist currently generates an independent `Info.plist` file, as such worth keeping the plist keys definitions consistent rather than mixing the setting style and plist file style.

Perhaps an independent exploration can be done to see how to add an option to switch between both, and what the limitations of the either approach is (e.g. are there keys that can only be defined in one but not the other? e.g. custom keys?)

**Review by kwridan — COMMENTED** [review]
Thanks @TheInkedEngineer

File: Sources/ProjectDescription/SettingsTransformers.swift (hunk 4 above)
Comment (TheInkedEngineer) [reply Sources/ProjectDescription/SettingsTransformers.swift:220]: As per the documentation of this key found [here](https://developer.apple.com/documentation/xcode/build-settings-reference#Application-Category) this has no impact on the `.plist`. If the user sets `GENERATE_INFOPLIST_FILE` to `YES` then at that point it should be reflected in the `.plist`.

I assume what you guys mean is that now if the user opt to add `GENERATE_INFOPLIST_FILE` this won't be reflected in the `.plist` generated by Tuist.

We can:

1. Document this and say that this will be added to the `Identity section` but not the `.plist` even with `GENERATE_INFOPLIST_FILE` set to `YES` and then work on it in a separate PR (I can start with it this week)
2. Delete the whole PR altogether 🥲

If there is anything else I can clear up please let me know.

Comment (danyf90) [reply Sources/ProjectDescription/SettingsTransformers.swift:220]: My point is that, given we are not supporting the GENERATE_INFOPLIST_FILE in any other way with any other helper, I wouldn' add this helpers here, but I would rather add them to [the current way of defining the plist](https://github.com/tuist/tuist/blob/main/Sources/ProjectDescription/InfoPlist.swift), probably as an extension on dictionaries of [String: InfoPlist.Value]?

Comment (TheInkedEngineer) [reply Sources/ProjectDescription/SettingsTransformers.swift:220]: I will create a fresh project (without tuist) play around with those values, see how everything changes, document it and then we can make a better decision. I'll keep you guys posted.

Comment (TheInkedEngineer) [reply Sources/ProjectDescription/SettingsTransformers.swift:220]: Creating a new iOS project (without Tuist):
- `GENERATE_INFOPLIST_FILE` is by default set to `YES`.
-  `INFOPLIST_KEY_LSApplicationCategoryType` adds nothing to the `.plist` file. But modifies the Identity tab under General. And in the `.pbxproj` adds `INFOPLIST_KEY_LSApplicationCategoryType = "public.app-category.education";` under `buildSettings` for Release and Debug.

There is a better explanation on the new management system found here: https://juejin.cn/post/7197361396219772983
Page is in Chinese but I translated the page to English.

Copied form that article: 

```
Build Settings - Generate Info.plist File (GENERATE_INFOPLIST_FILE) This field is newly introduced by Xcode13, it indicates whether to enable the new mode of generating Info.plist file, YES starts, NO does not enable. This field defaults to YES for new projects, and NO for old projects.

When the value is YES, Xcode uses the new Info.plist file generation mode. Xcode will take data from the Build Setting - Info.plist Values ​​module and Info - Custom iOS Target Properties to generate the final Info.plist file when packaging .
When the value is NO, Xcode uses the old Info.plist file generation mode. We only need to configure parameters in the Info.plist physical file in the project as before, and Xcode will read the configuration in the Info.plist physical file to generate the Info.plist file in the final package body when packaging.

First of all, let's make it clear: the name of the App is finally determined by the CFBundleDisplayName field in the Info.plist file in the compiled package body . If there is no CFBundleDisplayName field in the Info.plist, the value of the CFBundleName field will be taken.
Due to the new features of Xcode14, the Display Name set in the Xcode panel will only be synchronized to Build Setting - Info.plist Values ​​- Bundle Display Name .
Since we used the old Info.plist file mode (that is, the value of BuildSetting - Generate Info.plist File is NO), the settings of BuildSetting - Info.plist Values ​​- Bundle Display Name will not be synchronized to the final generated package body Info.plist file. This indirectly leads to the problem that setting the Display Name in the Xcode panel does not take effect.
```

So basically long story short this is a whole new `.plist` management system.

I assume this PR is a half solution, and should be addressed in a bigger way?
What would be the best next steps then?

Comment (danyf90) [reply Sources/ProjectDescription/SettingsTransformers.swift:220]: Discussing [here](https://tuistapp.slack.com/archives/CCXC295QS/p1678896591526059?thread_ts=1678697609.654359&cid=CCXC295QS)
